src/olympia/versions/models.py

```diff
@@ -635,6 +635,17 @@ def get_background_images_encoded(self, header_only=False):
             for name, background in utils.get_background_images(
                 file_obj, theme_data=None, header_only=header_only).items()}
 
+    def can_be_disabled_and_deleted(self):
+        addon = self.addon
+        if self.recommendation_approved and self == addon.current_version:
+            qs = addon.versions.exclude(id=self.id)
+            previous_version = qs.latest() if qs.exists() else None
+            previous_approved = (
+                previous_version and previous_version.recommendation_approved)
```

Comment:
mmm, there are 2 things that strike me as potentially problematic here:
- Not filtering on `channel`. While unlisted versions should never have `recommendation_approved` set currently, it would be good to make sure now that we are future proof.
- More worryingly, this cannot return a version that is already deleted because you're using `addon.versions`, but this could return a version that is already disabled.

Reply:
> this cannot return a version that is already deleted because you're using addon.versions, but this could return a version that is already disabled.

not returning deleted versions is okay (and what we want); disabled versions should probably be excluded.

Reply:
yeah my sentence was unclear: we clearly don't want deleted versions, I was highlighting the fact that `addon.versions` buys you this, but not excluding disabled ones.